### PR TITLE
BREAKING: Replace Discord.Py with Interactions

### PR DESCRIPTION
**Pull request by MatejMecka**
Oh boy, we haven't maintained this in a year. So this is what happens in this PR.

- Replace Discord.Py with Interactions.Py

The reasoning behind this was we were using for some of our commands. Turns out that Discord.Py and Interactions.Py are incompatible together so one had to go. This PR rewrote all the Discord interactions to use Interactions.Py and remove Discord.Py where used.

- Update Dependencies

The second thing that has been done was the requirements.txt file was rebuilt in order to use the latest features of all the dependencies we have. I don't believe we have anything breaking here.

### CODE REVIEW

File: test/test_stellar.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def test_fetch_last_tx(self, m: requests_mock.Mocker):
             json=self.TRANSACTIONS,
         )
         last_tx = fetch_last_tx()
-        self.assertEqual(last_tx, datetime(2021, 5, 20))
```
Comment (Smephite) [inline test/test_stellar.py:59]: This actually has to stay the same as we are comparing against a hard coded reference (`2021-05-20T13:49:32Z`)

File: main.py [hunk 2]
```diff
@@ -48,17 +45,18 @@ def processVote(message_id, author, backer):
 
 @client.event
 async def on_ready():
-    await client.change_presence(status=discord.Status.offline)  # Let's hide the bot
-    print(f"We have logged in as {client.user}")
+    # Let's hide the bot
```
Comment (Smephite) [inline main.py:48]: No more hiding?

File: test/test_stellar.py (hunk 1 above)
Comment (MatejMecka) [reply test/test_stellar.py:59]: Noted! Updating this :)

File: helpers/discord.py [hunk 3]
```diff
@@ -36,11 +36,13 @@ async def leaderboard(conn, client, message, limit):
...
+    channel = await message.get_channel() 
+    await channel.send("And the results are in!", embeds=[embed], files=discord_file)
 
 
 def hasRole(roles, REQUIRED_ROLE_ID):
-    if discord.utils.get(roles, id=int(REQUIRED_ROLE_ID)):
+    roles = [int(role) for role in roles]
+    if REQUIRED_ROLE_ID in roles:
         return True
     return False
```
[the remark this answers is not in the document]
Comment (MatejMecka) [reply helpers/discord.py:47]: Makes Sense haha

File: main.py [hunk 4]
```diff
@@ -130,4 +129,4 @@ async def _my_pub_key(ctx):
 
 if __name__ == "__main__":
     setup_db(conn)
-    client.run(DISCORD_BOT_TOKEN)
+    client.start()
```
[the remark this answers is not in the document]
Comment (MatejMecka) [reply main.py:132]: With Interactions you specify it in the client: https://discord-py-slash-command.readthedocs.io/en/latest/quickstart.html#first-let-s-run-the-bot

File: main.py (hunk 2 above)
Comment (MatejMecka) [reply main.py:48]: Moved into the Client: 

```py
client = Client(token=DISCORD_BOT_TOKEN, intents=Intents.GUILD_MESSAGES | Intents.DIRECT_MESSAGES |
                Intents.GUILD_MESSAGE_REACTIONS | Intents.GUILD_MESSAGE_CONTENT, presence=ClientPresence(status=StatusType.OFFLINE))
```

**Review by Smephite — APPROVED** [review]
LGTM, didn't do deployment tests tho.

**Pull request merged**